Automattic/jetpack
# Connection: always display connection button on connection screen

If connection fails, the user might end up on the connection screen with connection button hidden.
This PR will make connection button show up no matter what.
Affected plugins: Jetpack, VaultPress Backup.

## Proposed changes:
* Always display connection button on connection screen.

### Other information:


## Jetpack product discussion
1204656910285034-as-1204656907963581

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
Before checking out the branch, please try to reproduce the issue to confirm it's really fixed.

Please install [this plugin](https://gist.github.com/sergeymitr/9d60ccac69e70b0a2fca4e731e26ed61) to make Jetpack require full connection.
This is the way of WoA, although we won't be able to fully test the changes there, so this is the workaround.

1. Install Jetpack, connect it.
2. Install VaultPress Backup, Jetpack Debug plugins.
3. Use "Jetpack Debug -> Broken Token" to randomize primary user ID.
4. Go to Jetpack Dashboard, you should see the connection screen with no "Set up Jetpack" button.
5. Go to "Jetpack -> VaultPress Backup" page, confirm the connection button doesn't show up.
6. Checkout the branch, rebuild `plugins/jetpack` and `packages/backup` if needed.
7. Go to Jetpack Dashboard, confirm that the "Set up Jetpack" button shows up.
8. Go to "Jetpack -> VaultPress Backup" page, confirm the connection button shows up.

_Please note that connection flow will still not work in this scenario and fail further down the path. It will be addressed separately.
The goal of this PR is to prevent the situations where connection screen shows up and tells you to click the "Set up Jetpack" button, but the button is not there._
